cds/modules/deposit/api.py

```diff
@@ -905,11 +906,74 @@ def _publish_edited(self):
 
         return super(Video, self)._publish_edited()
 
+    def _has_chapters_changed(self, old_record=None):
+        """Check if chapters in description have changed."""
+        current_description = self.get("description", "")
+        current_chapters = parse_video_chapters(current_description)
+
+        if old_record is None:
+            # First publish - trigger if chapters exist
+            return len(current_chapters) > 0
+
+        old_description = old_record.get("description", "")
+        old_chapters = parse_video_chapters(old_description)
+
+        # Compare chapter timestamps and titles
+        if len(current_chapters) != len(old_chapters):
+            return True
+
+        for curr, old in zip(current_chapters, old_chapters):
+            if curr["seconds"] != old["seconds"] or curr["title"] != old["title"]:
+                return True
+
+        return False
+
+    def _trigger_chapter_frame_extraction(self):
+        """Trigger chapter frame extraction asynchronously for existing video files."""
+        try:
+            # Get the current flow for this deposit
+            current_flow = FlowMetadata.get_by_deposit(self["_deposit"]["id"])
+
+            if current_flow is None:
+                current_app.logger.warning(
+                    f"No current flow found for video {self.id}. Cannot trigger chapter frame extraction."
+                )
+                return
+
+            current_app.logger.info(
+                f"Triggering asynchronous ExtractChapterFramesTask for video {self.id} with flow {current_flow.id}"
+            )
+
+            payload = current_flow.payload.copy()
+
+            current_app.logger.info(f"Submitting ExtractChapterFramesTask with payload: {payload}")
+
+            ExtractChapterFramesTask().s(**payload).apply_async()
+
+            current_app.logger.info(
+                f"ExtractChapterFramesTask submitted asynchronously for video {self.id}, flow_id: {current_flow.id}"
+            )
+        except Exception as e:
+            current_app.logger.error(
+                f"Failed to trigger async chapter frame extraction for video {self.id}: {e}"
+            )
+            import traceback
+
+            current_app.logger.error(f"Traceback: {traceback.format_exc()}")
+
     @mark_as_action
     def publish(self, pid=None, id_=None, **kwargs):
         """Publish a video and update the related project."""
         # save a copy of the old PID
         video_old_id = self["_deposit"]["id"]
+
+        # Check if this is a republish and get the old record
+        old_record = None
+        try:
```

Comment:
I think here we can avoid the extra query to the DB.... We need to just save the old description so we can compare it again after the publish to check if the chapters have been changed...

something like

```suggestion
        old_description = self["description"]
```